packages/app/src/cli/services/transform.ts

```diff
@@ -0,0 +1,258 @@
+import {AbortError} from '@shopify/cli-kit/node/error'
+import fs from 'fs'
+import {fileURLToPath} from 'url'
+
+const packages = ['@shopify/polaris-codemods']
+
+interface TransformOptions {
+  /** Files or directory to transform */
+  include?: string
+
+  /** The package to run transforms for, \@scope/package. */
+  package?: string
+
+  /** The name of transform */
+  transform?: string
+
+  /** If true, no code will be edited */
+  dry?: boolean
+
+  /** Show more information about the transform process */
+  verbose?: boolean
+
+  /** If true, the changed output is printed for comparison */
+  print?: boolean
+
+  /** If true, bypass Git safety checks and forcibly run transform */
+  force?: boolean
+}
+
+export async function transform(options: TransformOptions) {
+  if (!((await isClean(cwd())) || options.dry)) {
+    if (options.force) {
+      renderWarning({
+        headline: 'Forcibly continuing',
+        body: 'You have opted to bypass the Git safety check. This may result in lost work.',
+      })
+    } else {
+      renderWarning({
+        headline: 'Before we continue, please stash or commit your git changes.',
+        body: 'You may use the --force flag to override this safety check.',
+      })
+      return
+    }
+  }
+
+  const pluginsPath = joinPath(dirname(fileURLToPath(import.meta.url)), 'node_modules')
+  const packageManager = new PluginManager({
+    cwd: cwd(),
+    pluginsPath,
+  })
+
+  const npmPackageString =
+    options.package ??
+    // Use the `@shopify/polaris-codemods` package as a default for now.
+    // This could be removed to prompt a list of supported codemod packages
+    '@shopify/polaris-codemods' ??
+    (await renderAutocompletePrompt({
+      message: 'Select a package',
+      choices: packages.map((pkg) => {
+        return {value: pkg, label: pkg}
+      }),
+    }))
+
+  const config = await fetchPackageConfig(npmPackageString, packageManager)
+  const transforms = {...config.transforms, ...config.presets}
+
+  const transform =
+    options.transform ??
+    (await renderAutocompletePrompt({
+      message: 'Select a transform to apply',
+      choices: Object.keys(transforms).map((transformName) => {
+        return {
+          value: transformName,
+          label: transformName,
+          group: transformName in config.transforms! ? 'Transforms' : 'Presets',
+        }
+      }),
+    }))
+
+  const transformPath = Object.entries(transforms).find(([id]) => {
+    return transform === id
+  })?.[1]
+
+  if (!transformPath || !fs.existsSync(transformPath)) {
+    throw new AbortError(`No transform found for ${options.transform}`)
+  }
+
+  const transformInfo: TransformInfo = await import(transformPath)
+  const extensions = transformInfo.extensions ?? ['js', 'jsx', 'ts', 'tsx']
+  const includePath = options.include ?? './'
+  const globPattern = joinPath(includePath, `**/*.(${extensions.join('|')})`)
+
+  const filePaths = await glob(globPattern, {cwd: cwd()})
```

Comment:
To be clear, you want this to run in the actual cwd from which the command is run, right?

Reply:
Good point. I think this is what jscodeshift expects, but I will test it from a few sub directories to make sure the paths are correct.